examples/tabular/example_tabular_predictor_v2.py

```diff
@@ -0,0 +1,62 @@
+
+from autogluon.tabular import TabularPrediction as task
+from autogluon.tabular.task.tabular_prediction.predictor import TabularPredictor
+from autogluon.tabular.task.tabular_prediction.predictor_v2 import TabularPredictorV2
+
+
+################
+# Loading data #
+################
+
+train_data = task.Dataset(file_path='https://autogluon.s3.amazonaws.com/datasets/Inc/train.csv')
+test_data = task.Dataset(file_path='https://autogluon.s3.amazonaws.com/datasets/Inc/test.csv')
+label = 'class'
+hyperparameters = {'RF': {}}
+train_data = train_data.head(1000)  # subsample for faster demo
+
+##################################
+# Fitting with the old Predictor #
+##################################
+
+predictor1 = task.fit(
+    train_data=train_data,
+    label=label,
+    hyperparameters=hyperparameters,
+)
+predictor1.leaderboard(test_data)
+
+##################################
+# Fitting with the new Predictor #
+##################################
+
+predictor2 = TabularPredictorV2(label=label)
+
+# Showing that the new Predictor can be saved and loaded prior to calling .fit()
+output_dir = predictor2.output_directory
+predictor2.save()
```

Comment:
an open question I would like to discuss is the serialization of TabularPredictor, so for now my understanding it that it's using a directory to save possibly multiple models and to reduce memory cost as you can dump intermediate buffers on disk. 

However, a directory isn't a good choice for serialization, as it's vulnerable(integrity, file system permissions, ...), cannot be safely copied directly or through communications. In constrast, a binary file can be pickled, base64 encoded and passed through https request, or directly shared through queues or pipes.

Any thought on this before we dig into more complex use cases, e.g., multi model prediction tasks?

Reply:
A simple solution is to use the zipfile to group the artifacts.

Reply:
Agreed that zipfile or binary file would be the ideal format for transporting the predictor. Will plan to address this in Part 2/Part 3 as a follow-up